lib/zfs.py

```diff
@@ -55,35 +54,34 @@ def size(self):
         # Return a tuple of used and available for this pool.
         return self.used, self.total
 
-    def add(self, devices):
-        '''Add devices to the zfs storage pool. This operation can fail if
-        the number of disks does not match the origin raidz pool.'''
-        # The self.mount_point will only exist if we created the zfs pool.
-        if self.mountpoint:
-            # The command that adds a device to a zfs pool.
-            cmd = 'sudo zpool add {0} raidz '.format(self.pool_name)
+    def add(self, devices, force=False):
+        '''Add devices to a raidz zfs storage pool. This operation can fail if
+        the number of disks does not match the original raidz pool.'''
+        force_flag = '-f' if force else ''
+        # When the pool exists, add devices to the pool.
+        if ZfsPool.exists(self.pool_name):
+            # The command that adds a device to a raidz zfs pool.
+            cmd = 'sudo zpool add {0} {1} raidz '.format(force_flag,
+                                                         self.pool_name)
             cmd += ' '.join(devices)
             print(cmd)
             check_call(split(cmd))
-        else:
-            # The command that creates a zfs pool without a mount point.
-            cmd = 'sudo zpool create {0} raidz '.format(self.pool_name)
+            self.devices.append(devices)
+        else:  # The pool does not yet exist, so we must create a new pool.
```

Comment:
It might be better to have a `_zpool_create` and `_zpool_add` private class methods to make unit testing easier and condense the complexity of this method.